### PR TITLE
Add table oci_mysql_configuration_custom. Closes #274

### LINKED ISSUE #274: Add table oci_mysql_configuration_custom
**References**
https://github.com/turbot/steampipe-plugin-oci/pull/255

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro oci-test % ./tint.js oci_mysql_configuration_custom
Test names: [ 'tests/oci_mysql_configuration_custom' ]
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_mysql_configuration_custom []

PRETEST: tests/oci_mysql_configuration_custom

TEST: tests/oci_mysql_configuration_custom
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create
 <= read (data resources)

Terraform will perform the following actions:

  # data.local_file.configuration will be read during apply
  # (config refers to values not yet known)
 <= data "local_file" "configuration"  {
      + content        = (known after apply)
      + content_base64 = (known after apply)
      + filename       = "/private/var/folders/df/957ty4t146vfy54wm4254kp40000gp/T/tests/oci_mysql_configuration_custom/terraform/test/output.json"
      + id             = (known after apply)
    }

  # null_resource.named_test_resource will be created
  + resource "null_resource" "named_test_resource" {
      + id = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + lifecycle_state = (known after apply)
  + region          = "ap-mumbai-1"
  + resource_id     = (known after apply)
  + resource_name   = (known after apply)
  + shape_name      = (known after apply)
  + tenancy_ocid    = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
  + type            = (known after apply)
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci mysql configuration create --compartment-id ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq --shape-name VM.Standard.E2.1 --output json > /private/var/folders/df/957ty4t146vfy54wm4254kp40000gp/T/tests/oci_mysql_configuration_custom/terraform/test/output.json"]
null_resource.named_test_resource: Creation complete after 2s [id=3919732474623265274]
data.local_file.configuration: Reading...
data.local_file.configuration: Read complete after 0s [id=51088cf14806e0d514366b23decbb2c28031ebd9]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

lifecycle_state = "ACTIVE"
region = "ap-mumbai-1"
resource_id = "ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaqfhwvgwvxjw37b6bfzx4qmktzse6tb46aq37eth2fvoug677d6ea"
resource_name = "mysqlconfiguration20210824090628"
shape_name = "VM.Standard.E2.1"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
type = "CUSTOM"

Running SQL query: test-get-query.sql

[
  {
    "display_name": "mysqlconfiguration20210824090628",
    "id": "ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaqfhwvgwvxjw37b6bfzx4qmktzse6tb46aq37eth2fvoug677d6ea",
    "lifecycle_state": "ACTIVE",
    "shape_name": "VM.Standard.E2.1",
    "type": "CUSTOM"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "display_name": "mysqlconfiguration20210824090628",
    "id": "ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaqfhwvgwvxjw37b6bfzx4qmktzse6tb46aq37eth2fvoug677d6ea",
    "lifecycle_state": "ACTIVE",
    "shape_name": "VM.Standard.E2.1",
    "type": "CUSTOM"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "region": "ap-mumbai-1",
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "mysqlconfiguration20210824090628"
  }
]
✔ PASSED

POSTTEST: tests/oci_mysql_configuration_custom
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # null_resource.destroy_test_resource will be created
  + resource "null_resource" "destroy_test_resource" {
      + id = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.
null_resource.destroy_test_resource: Creating...
null_resource.destroy_test_resource: Provisioning with 'local-exec'...
null_resource.destroy_test_resource (local-exec): Executing: ["/bin/sh" "-c" "oci mysql configuration delete --configuration-id ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaqfhwvgwvxjw37b6bfzx4qmktzse6tb46aq37eth2fvoug677d6ea --force"]
null_resource.destroy_test_resource: Creation complete after 0s [id=1868787879792005613]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

TEARDOWN: tests/oci_mysql_configuration_custom


SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  display_name,
  id,
  description,
  lifecycle_state as state,
  time_created
from
  oci_mysql_configuration_custom;
+----------------------------------+-------------------------------------------------------------------------------------------------------+-------------+---------+---------------------+
| display_name                     | id                                                                                                    | description | state   | time_created        |
+----------------------------------+-------------------------------------------------------------------------------------------------------+-------------+---------+---------------------+
| mysqlconfiguration20210727122951 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaz2rl7gidlvihyhuljvdlaiqrsjdcfxa3ntdmluvhncguz2yacxda | <null>      | DELETED | 2021-07-27 12:29:51 |
| mysql20210729160050              | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaa4byx63at5pimfybveoouevzqh7b3gqpcnb5cy2kjo4ardxdbkmiq | <null>      | DELETED | 2021-07-29 10:30:59 |
| test1234                         | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaajrbxtmn2f7xc5oycal4acr2hxd2rrvihxia255rf5f5vhbjtr6ha | testing     | DELETED | 2021-07-08 05:23:49 |
| mysqlconfiguration20210727122458 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaf46aywfgzdnxkjak3hgd5a6cbeowrmlpacqjhet6qaiwshvi6uhq | <null>      | DELETED | 2021-07-27 12:24:58 |
| mysqlconfiguration20210727115509 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaawa3d6nim6c6tfhauyanwt2qtn7u43ru533mc4vtse5fuhm7beqsq | <null>      | DELETED | 2021-07-27 11:55:09 |
| mysqlconfiguration20210727122102 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaanhsfpcshz4hbb7wjf3ij2ry45shk6ycl3levfoudkijqofnyj5ra | <null>      | DELETED | 2021-07-27 12:21:02 |
| mysqlconfiguration20210727123149 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaka7pgbosxniagxkxbyuasofssnfwzbifjphho4tvzgfezsagd32a | <null>      | DELETED | 2021-07-27 12:31:49 |
| mysqlconfiguration20210818101556 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaavpyzlerwcvmnwlk47qta545fw36dqqe3wr4cspiwls3k2sqh2isq | <null>      | ACTIVE  | 2021-08-18 10:15:56 |
| mysqlconfiguration20210818081915 | ocid1.mysqlconfiguration.oc1.iad.aaaaaaaayt7z3otdamxawk2woqyjjwdt3aanc63joonmljqapu4uoibfiroa         | <null>      | ACTIVE  | 2021-08-18 08:19:15 |
| mysqlconfiguration20210818101436 | ocid1.mysqlconfiguration.oc1.iad.aaaaaaaagwbycssq5ehecezdawphsjensbii3copn2vsaws5brtthkmuc7nq         | <null>      | ACTIVE  | 2021-08-18 10:14:36 |
| mysqlconfiguration20210818101059 | ocid1.mysqlconfiguration.oc1.iad.aaaaaaaardtn7gkaj7mfjbwz2xzd5lqhaqzpxs3lkxczwu7swigh6jtz3kga         | <null>      | ACTIVE  | 2021-08-18 10:10:59 |
| mysqlconfiguration20210818082409 | ocid1.mysqlconfiguration.oc1.iad.aaaaaaaapznaslgbutagxzw2oh5z2xsuig7chclga4ptvakakfhr6sfubjhq         | <null>      | ACTIVE  | 2021-08-18 08:24:09 |
+----------------------------------+-------------------------------------------------------------------------------------------------------+-------------+---------+---------------------+
> select
  display_name,
  id,
  description,
  lifecycle_state as state,
  time_created
from
  oci_mysql_configuration_custom 
where
  lifecycle_state = 'DELETED';
+----------------------------------+-------------------------------------------------------------------------------------------------------+-------------+---------+---------------------+
| display_name                     | id                                                                                                    | description | state   | time_created        |
+----------------------------------+-------------------------------------------------------------------------------------------------------+-------------+---------+---------------------+
| mysqlconfiguration20210727122951 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaz2rl7gidlvihyhuljvdlaiqrsjdcfxa3ntdmluvhncguz2yacxda | <null>      | DELETED | 2021-07-27 12:29:51 |
| mysql20210729160050              | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaa4byx63at5pimfybveoouevzqh7b3gqpcnb5cy2kjo4ardxdbkmiq | <null>      | DELETED | 2021-07-29 10:30:59 |
| test1234                         | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaajrbxtmn2f7xc5oycal4acr2hxd2rrvihxia255rf5f5vhbjtr6ha | testing     | DELETED | 2021-07-08 05:23:49 |
| mysqlconfiguration20210727122458 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaf46aywfgzdnxkjak3hgd5a6cbeowrmlpacqjhet6qaiwshvi6uhq | <null>      | DELETED | 2021-07-27 12:24:58 |
| mysqlconfiguration20210727115509 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaawa3d6nim6c6tfhauyanwt2qtn7u43ru533mc4vtse5fuhm7beqsq | <null>      | DELETED | 2021-07-27 11:55:09 |
| mysqlconfiguration20210727122102 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaanhsfpcshz4hbb7wjf3ij2ry45shk6ycl3levfoudkijqofnyj5ra | <null>      | DELETED | 2021-07-27 12:21:02 |
| mysqlconfiguration20210727123149 | ocid1.mysqlconfiguration.oc1.ap-mumbai-1.aaaaaaaaka7pgbosxniagxkxbyuasofssnfwzbifjphho4tvzgfezsagd32a | <null>      | DELETED | 2021-07-27 12:31:49 |
+----------------------------------+-------------------------------------------------------------------------------------------------------+-------------+---------+---------------------+
> 
```
</details>

### CODE REVIEW

File: docs/tables/oci_mysql_configuration.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Table: oci_mysql_configuration
+
+DB System configurations are collections of variables which define the operation of the MySQL DB System.
```
Comment (bigdatasourav) [inline docs/tables/oci_mysql_configuration.md:3]: Updated the description as I already provided.

File: docs/tables/oci_mysql_configuration_custom.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Table: oci_mysql_configuration_custom
+
+DB System configurations are collections of variables which define the operation of the MySQL DB System.
+
+Configurations have a default set of variables assigned to them, user and system variables. To add variables, you must create a new configuration with the desired variable definitions, or copy an existing configuration, edit it accordingly, and edit the DB System to use the new configuration.
```
Comment (bigdatasourav) [inline docs/tables/oci_mysql_configuration_custom.md:5]:
```suggestion
Configurations have a default set of variables assigned to them, user and system variables. To add variables, you must create a new configuration which will be a custom configuration with the desired variable definitions, or copy an existing configuration, edit it accordingly, and edit the DB System to use the new configuration.
```

File: docs/tables/oci_mysql_configuration.md (hunk 1 above)
Comment (bigdatasourav) [inline docs/tables/oci_mysql_configuration.md:3]: remove oci_mysql_configuration table changes from this PR.

File: oci/plugin.go [hunk 3]
```diff
@@ -102,6 +102,8 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"oci_logging_log_group":                                             tableLoggingLogGroup(ctx),
 			"oci_mysql_backup":                                                  tableMySQLBackup(ctx),
 			"oci_mysql_channel":                                                 tableMySQLChannel(ctx),
+			"oci_mysql_configuration":                                           tableMySQLConfiguration(ctx),
+			"oci_mysql_configuration_custom":                                    tableMySQLCustomConfiguration(ctx),
```
Comment (bigdatasourav) [inline oci/plugin.go:106]:
```suggestion
			"oci_mysql_configuration_custom":                                    tableMySQLConfigurationCustom(ctx),
```

File: oci/table_oci_mysql_configuration_custom.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+            Hydrate:    getCustomConfiguration,
+        },
+        List: &plugin.ListConfig{
+            Hydrate: listMySQLCustomConfigurations,
+        },
+        GetMatrixItem: BuildCompartementRegionList,
+        Columns: []*plugin.Column{
+            {
+                Name:        "display_name",
+                Description: "The display name of the Configuration.",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration_custom.go:32]:
```suggestion
                Description: "The display name of the configuration.",
```

File: oci/table_oci_mysql_configuration_custom.go [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+                Type:        proto.ColumnType_STRING,
+            },
+            {
+                Name:        "id",
+                Description: "The OCID of the Configuration.",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration_custom.go:37]:
```suggestion
                Description: "The OCID of the configuration.",
```

File: oci/table_oci_mysql_configuration_custom.go [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+                Type:        proto.ColumnType_STRING,
+                Transform:   transform.FromCamel(),
+            },
+            {
+                Name:        "parent_configuration_id",
+                Description: "The OCID of the Configuration from which this Configuration is derived.",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration_custom.go:43]:
```suggestion
                Description: "The OCID of the configuration from which this configuration is derived.",
```

File: oci/table_oci_mysql_configuration_custom.go [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+
+            // other columns
+            {
+                Name:        "description",
+                Description: "User-provided data about the Configuration.",
+                Type:        proto.ColumnType_STRING,
+            },
+            {
+                Name:        "shape_name",
+                Description: "The name of the associated Shape.",
```
Comment (bigdatasourav) [inline oci/table_oci_mysql_configuration_custom.go:68]:
```suggestion
                Description: "The name of the associated shape.",
```

**Review by bigdatasourav — CHANGES_REQUESTED** [review]
@c0d3r-arnab Please check the suggested changes

File: oci-test/tests/oci_mysql_configuration_custom/variables.tf [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  config_file_profile = var.config_file_profile
+}
+
+locals {
+  path = "${path.cwd}/output.json"
+}
+
+resource "null_resource" "named_test_resource" {
+  provisioner "local-exec" {
+    command = "oci mysql configuration create --compartment-id ${var.tenancy_ocid} --shape-name ${var.shape} --output json > ${local.path}"
```
Comment (bigdatasourav) [inline oci-test/tests/oci_mysql_configuration_custom/variables.tf:48]: Add Post-test to delete the resource as you are using CLI here, TF will not delete it.

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**